crates/sui-json-rpc/src/indexer_api.rs

```diff
@@ -425,11 +468,20 @@ impl<R: ReadApiServer> IndexerApiServer for IndexerApi<R> {
                 })?
                 .value;
 
-            Ok(Page {
-                data: vec![domain.to_string()],
-                next_cursor: None,
-                has_next_page: false,
-            })
+            let domain_name = domain.to_string();
+
+            let resolved_address = self
+                .resolve_name_service_address(domain_name.clone())
+                .await?;
+
+            if resolved_address.is_none() || resolved_address != Some(address) {
+                return Ok(result);
+            }
```

Comment:
The call to `resolve_name_service_address` is still necessary to trigger a potential error on expiry, but we may not need the further checks for `is_none` or `resolved_address != Some(address)` because the registry is now updated consistently on-chain.